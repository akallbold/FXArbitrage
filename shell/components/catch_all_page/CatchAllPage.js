import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { getErrorPageType, getErrorPageData } from 'components/catch_all_page'
import { getPrevPathname } from 'selectors/platform_selectors'
import * as pages from './CatchAll.pages'

export const CatchAllPage = ({ type, data, prevPathname }) => {
  const ErrorPage = pages[type]
  return <ErrorPage {...data} prevPathname={prevPathname} />
}

CatchAllPage.propTypes = {
  type: PropTypes.string,
  data: PropTypes.object,
  prevPathname: PropTypes.string
}

CatchAllPage.defaultProps = {
  type: pages.TYPE_404,
  data: {},
  prevPathname: null
}

const mapStateToProps = state => ({
  type: getErrorPageType(state),
  data: getErrorPageData(state),
  prevPathname: getPrevPathname(state)
})

const mapDispatchToProps = {

}

export default connect(mapStateToProps, mapDispatchToProps)(CatchAllPage)
