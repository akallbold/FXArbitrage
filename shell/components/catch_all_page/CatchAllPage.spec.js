import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { CatchAllPage } from './CatchAllPage'
import { TYPE_404 } from './CatchAll.pages'

describe('<CatchAllPage />', () => {
  it('should render properly given all props', () => {
    const component = shallow(
      <CatchAllPage
        type={TYPE_404}
        prevPathname='some/path/here'
      />
    )
    expect(toJson(component)).toMatchSnapshot()
  })
})
