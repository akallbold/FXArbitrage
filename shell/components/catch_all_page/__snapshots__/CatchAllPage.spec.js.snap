// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CatchAllPage /> should render properly given all props 1`] = `
<Page404
  prevPathname="some/path/here"
/>
`;
